src/modules/formBuilder/components/FormEditorItemProperties.tsx

```diff
@@ -0,0 +1,182 @@
+import { Stack, Typography } from '@mui/material';
+import LabeledCheckbox from '@/components/elements/input/LabeledCheckbox';
+import TextInput from '@/components/elements/input/TextInput';
+import FormSelect from '@/modules/form/components/FormSelect';
+import { isPickListOption } from '@/modules/form/types';
+import {
+  localResolvePickList,
+  MAX_INPUT_AND_LABEL_WIDTH,
+} from '@/modules/form/util/formUtil';
+import { FormItem } from '@/types/gqlTypes';
+
+const dataCollectedAboutPickList =
+  localResolvePickList('DataCollectedAbout') || [];
+const disabledDisplayPickList = localResolvePickList('DisabledDisplay') || [];
+const inputSizePickList = localResolvePickList('InputSize') || [];
+
+interface FormEditorItemPropertiesProps {
+  item: FormItem;
+  onChangeProperty: (attr: keyof FormItem, newProperty: any) => void;
+}
+
+const FormEditorItemProperties: React.FC<FormEditorItemPropertiesProps> = ({
+  item,
+  onChangeProperty,
+}) => {
+  return (
+    <>
+      <Typography>Properties</Typography>
+      <Stack
+        gap={2}
+        component='form'
+        onSubmit={() => {
+          console.log('submitted'); // TODO(#6103)
+        }}
+        sx={{ maxWidth: MAX_INPUT_AND_LABEL_WIDTH }}
+      >
+        <TextInput
+          label='Link ID'
+          value={item.linkId}
+          onChange={(e) => {
+            onChangeProperty('linkId', e.target.value);
+          }}
+          helperText='Unique ID for this form item'
+        />
+        <LabeledCheckbox
```

Comment:
Suggested improvements (in this pr or another)
* maybe `helperText="If checked, this date will be recorded as the Assessment Date on the assessment"`
* only show if type is date, and if form type is one of `AssessmentRole`s. I think there will be other cases where the configurability depends on the `role` of the definition, so we want to thread that data through props

Reply:
Ok, sounds good. Let me know if you think just passing the whole entire `definition` object is too much data -- I was thinking we will need it later for the sake of conditional logic, for instance, because we need to find out what other link IDs exist in this form.